java/src/jni/h5dImp.c

```diff
@@ -1315,54 +1728,15 @@ H5DreadVL_asstr(JNIEnv *env, hid_t did, hid_t tid, hid_t mem_sid, hid_t file_sid
     return status;
 }
 
-/**
- *  Read VLEN data into array of arrays.
- *  Object[] buf contains VL arrays of data points
- *  Currently only deal with variable length of atomic data types
- */
-/* old version */
-
 /*
  * Class:     hdf_hdf5lib_H5
- * Method:    H5Dread_VLStrings
- * Signature: (JJJJJ[Ljava/lang/String;)I
- */
-JNIEXPORT jint JNICALL
-Java_hdf_hdf5lib_H5_H5Dread_1VLStrings(JNIEnv *env, jclass clss, jlong dataset_id, jlong mem_type_id,
-                                       jlong mem_space_id, jlong file_space_id, jlong xfer_plist_id,
-                                       jobjectArray buf)
-{
-    htri_t isVlenStr = 0;
-    herr_t status    = FAIL;
-
-    UNUSED(clss);
-
-    if (NULL == buf)
-        H5_NULL_ARGUMENT_ERROR(ENVONLY, "H5Dread_VLStrings: read buffer is NULL");
-
-    if ((isVlenStr = H5Tdetect_class((hid_t)mem_type_id, H5T_STRING)) < 0)
-        H5_LIBRARY_ERROR(ENVONLY);
-
-    if (isVlenStr) {
-        if ((status = H5DreadVL_str(env, (hid_t)dataset_id, (hid_t)mem_type_id, (hid_t)mem_space_id,
-                                    (hid_t)file_space_id, (hid_t)xfer_plist_id, buf)) < 0)
-            H5_LIBRARY_ERROR(ENVONLY);
-    }
-    else
-        H5_BAD_ARGUMENT_ERROR(ENVONLY, "H5Dread_VLStrings: datatype is not variable length String");
-
-done:
-    return (jint)status;
-} /* end Java_hdf_hdf5lib_H5_H5Dread_1VLStrings */
-
-/*
- * Class:     hdf_hdf5lib_H5
- * Method:    H5DwriteVL
+ * Method:    H5Dwrite_VLStrings
  * Signature: (JJJJJ[Ljava/lang/String;)I
  */
 JNIEXPORT jint JNICALL
-Java_hdf_hdf5lib_H5_H5DwriteVL(JNIEnv *env, jclass clss, jlong dataset_id, jlong mem_type_id,
-                               jlong mem_space_id, jlong file_space_id, jlong xfer_plist_id, jobjectArray buf)
+Java_hdf_hdf5lib_H5_H5Dwrite_1VLStrings(JNIEnv *env, jclass clss, jlong dataset_id, jlong mem_type_id,
```

Comment:
With the new implementation of support for VL types, is it possible to coalesce these two paths for VL types into a single H5D(A)writeVL? It'd be nice to not have to change which routine you call based on the datatype anymore.

Reply:
Not really, strings are a different animal currently with existing code (which I probably broke with the correction).

Reply:
Seems like it should be incorporated in the near future.